src/material_pool.hpp

```diff
@@ -15,6 +15,54 @@ struct aiMaterial;
 
 namespace wr
 {
+	class Material
+	{
+	public:
+
+		Material();
+		Material(TextureHandle albedo, 
+				 TextureHandle normal, 
+				 TextureHandle rough_met, 
+				 TextureHandle ao, 
+				 bool alpha_masked = false, 
+				 bool double_sided = false);
+
+		Material(const Material& rhs) = default;
+
+		~Material();
+
+		TextureHandle& Albedo() { return m_albedo; }
+		void SetAlbedo(TextureHandle albedo);
+
+		TextureHandle& Normal() { return m_normal; }
+		void SetNormal(TextureHandle normal);
+
+		TextureHandle& RoughnessMetallic() { return m_rough_metallic; }
+		void SetRoughMetallic(TextureHandle rough_met);
+
+		TextureHandle& AmbientOcclusion() { return m_rough_metallic; }
+		void SetAmbientOcclusion(TextureHandle ao);
+
+		bool IsAlphaMasked() { return m_alpha_masked; }
+		void SetAlphaMasked(bool alpha_masked) { m_alpha_masked = alpha_masked; }
+
+		bool IsDoubleSided() { return m_double_sided; }
```

Comment:
Currently probably unused. So comment them out or mark them as deprecated. Having this code available will just give the impression we support it.

Reply:
Done 👍